### PR TITLE
Add interface for opening external urls (react-native).

### LINKED ISSUE #285: Provide option to open external application
### Description

There is a need to have an option for opening external application from within MagicScript Components app. Something like: Platform.open("external_app_schema").

---

Sample code:
```
onButtonClick = () => {
    Platform.openUrl('https://www.magicleap.com');
}
```

### JIRA

MXSCOMP-218 (React-native)

### Completion status
- [ ] [Android](https://github.com/magic-script/magic-script-components-react-native/issues/523)
- [ ] [iOS](https://github.com/magic-script/magic-script-components-react-native/issues/523)
- [ ] Lumin

### PR DESCRIPTION
**Pull request by pleszkiewicz**
Thank you for your contribution to MagicScript Project.
Before submitting this PR, please make sure:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests
- [ ] You have updated documentation

closes #285 

helio

### CODE REVIEW

File: src/platform.js [hunk 1]
```diff
@@ -3,6 +3,22 @@ class Platform {
         Platform.OS = platform.name();
         Platform.Version = platform.version();
     }
+
+    static setLinking(linking) {
+      Platform.linking = linking;
+    }
+
+    static canOpenUrl(url) {
```
Comment (kpiascik) [inline src/platform.js:11]: doesn't this need to be async?

File: src/platform.js [hunk 2]
```diff
@@ -3,6 +3,22 @@ class Platform {
...
+      try {
+        return Platform.linking.canOpenURL(url);
+      } catch {
+        return new Promise((resolve, reject) => reject({ message: 'Invalid format.' }));
+      }
+    }
+
+    static openUrl(url) {
```
Comment (kpiascik) [inline src/platform.js:19]: same here?

File: src/platform.js (hunk 1 above)
Comment (pleszkiewicz) [reply src/platform.js:11]: This function returns a promise so it does not need to be async. We can make it async, but we will need to change their body - sth like this:
```
static async canOpenUrl(URL) {
    try {
        const result = await Platform.linking.canOpenURL(url);
        return result;
    } catch {
        throw new Error('Invalid format.');
    }
}
```
Check [this page](https://javascript.info/async-await).

File: src/platform.js (hunk 2 above)
Comment (pleszkiewicz) [reply src/platform.js:19]: I think we don't need `async` here. Check [this](https://javascript.info/async-await).

**Pull request merged**